hudi-common/src/main/java/org/apache/hudi/io/storage/HoodieAvroFileWriterFactory.java

```diff
@@ -106,4 +106,14 @@ protected HoodieFileWriter newOrcFileWriter(
         config.getLong(HoodieStorageConfig.ORC_FILE_MAX_SIZE), filter);
     return new HoodieAvroOrcWriter(instantTime, path, orcConfig, schema, taskContextSupplier);
   }
+
+  private HoodieAvroWriteSupport getHoodieAvroWriteSupport(Configuration conf, Schema schema,
+                                                           HoodieConfig config, boolean enableBloomFilter) {
+    Option<BloomFilter> filter = enableBloomFilter ? Option.of(createBloomFilter(config)) : Option.empty();
+    HoodieAvroWriteSupport writeSupport = (HoodieAvroWriteSupport) ReflectionUtils.loadClass(
```

Comment:
Wondering if we should introduce a new abstract class `HoodieBaseAvroWriteSupport` (`HoodieAvroWriteSupport` will extend `HoodieBaseAvroWriteSupport`) and use that here, instead of directly using `HoodieAvroWriteSupport`.  So other custom implementation may extend `HoodieBaseAvroWriteSupport` or reuse part of `HoodieAvroWriteSupport` if needed.  Right now it has to extend `HoodieAvroWriteSupport`.   @suryaprasanna @vinothchandar  wdyt?

Reply:
AvroWriteSupport which is from the parquet-avro library already has a base implementation. We are over-ridding that class with HoodieAvroWriteSupport to include bloom filter and some footer metadata. 
So, will it be confusing to users, if we call this class a base class and do not provide any other implementation for this class on the repository?

Reply:
I'm thinking we can keep `HoodieBaseAvroWriteSupport` abstract and still provide the actual implementation in `HoodieAvroWriteSupport`.  This is more cosmetic.  I'm still ok with the current approach.